src/index.js

```diff
@@ -0,0 +1,73 @@
+/* eslint-env browser */
+// import memoize from "lodash.memoize";
+import isFinite from "lodash.isfinite";
+
+const globalCanvas = document.createElement("canvas");
+
+const isNumericOnly = (string) => /^[0-9.,]+$/.test(string);
+
+const parseUnitless = (value) => {
+  if (isFinite(value)) {
+    return value;
+  }
+  if (isNumericOnly(value)) {
+
+    return parseFloat(value);
+  }
+  return null;
+};
+
+const parseUnit = (value) => {
+  // http://stackoverflow.com/questions/2868947/split1px-into-1px-1-px-in-javascript
+  const matches = value.match(/^(\d+(?:\.\d+)?)(.*)$/);
+  const [, number, unit] = matches;
+  return { number: parseFloat(number), unit };
+};
+
+const calculateHeight = (size, lineHeight) => {
+  const {
+    number: sizeNumber,
+    unit: sizeUnit
+  } = parseUnit(size);
+
+  // If the line-height is unitless,
+  // multiply it by the font size.
+  const unitless = parseUnitless(lineHeight);
+  if (unitless) {
+    return `${sizeNumber * unitless}${sizeUnit}`;
+  }
+
+  // Otherwise, it's as simple as
+  // returning the line height.
+  return lineHeight;
+};
+
+const DEFAULT_FONT_WEIGHT = 400;
+const DEFAULT_FONT_STYLE = "normal";
+
+const measureText = ({
+  text,
+  fontFamily,
+  fontSize,
+  lineHeight,
+  fontWeight = DEFAULT_FONT_WEIGHT,
+  fontStyle = DEFAULT_FONT_STYLE,
+  canvas: userCanvas
```

Comment:
Possible to use the same default syntax here?

`canvas = globalCanvas`

then we can ditch line 57.